File: Linked_List/GFG_2. Add Two Numbers(non-reverse).py
class Solution:
    #Function to add two numbers represented by linked list.
    def addTwoLists(self, first, second):
        def reverse1(head):
            pre=None
            nex=None
            while head!=None:
                nex=head.next
                head.next=pre
                pre=head
                head=nex
            return pre
        
        first=reverse1(first)
        second=reverse1(second)
        dummy=Node(0)
        temp=dummy
        carry=0
        while first or second or carry>0:
            addition=0
            if first is not None:
                addition+=first.data
                first=first.next
                
            if second is not None:
                addition+=second.data
                second=second.next
            addition+=carry
            carry=addition//10
            if carry>0:
                addition=addition%10
            temp.next=Node(addition)    
            temp=temp.next
            
        return reverse1(dummy.next)

File: Linked_List/Java/GFG_2. Add Two Numbers(non-reverse).java
class Solution{
    
    public static Node reverse(Node head)
    {
     
        Node itr=head;
        Node prev=null;
        Node nex=null;

        while(itr!=null)
        {
            nex=itr.next;
            itr.next=prev;
            prev=itr;
            itr=nex;
        }
        return prev;
    }
    
    //Function to add two numbers represented by linked list.
    static Node addTwoLists(Node first, Node second){
            Node l1=reverse(first);
            Node l2=reverse(second);
            Node dummy=new Node(0);
            Node temp=dummy;
            int carry=0;
            while(l1!=null || l2!=null || carry==1)
            {
                int addition=0;
                if(l1!=null)
                {
                    addition+=l1.data;
                    l1=l1.next;
                }
                if(l2!=null)
                {
                    addition+=l2.data;
                    l2=l2.next;
                }
                addition+=carry;
                carry=addition/10;
                if(carry>0)
                {
                    addition=addition%10;
                }
               
                Node node = new Node(addition);
                temp.next=node;
                temp=temp.next;
           
            }
                return reverse(dummy.next);
     }
}